src/canopy/knowledge_base/record_encoder/openai.py

```diff
@@ -27,25 +36,31 @@ def __init__(self,
                         Defaults to 400.
             **kwargs: Additional arguments to pass to the underlying `pinecone-text. OpenAIEncoder`.
         """  # noqa: E501
-        encoder = OpenAIEncoder(model_name, **kwargs)
+        try:
+            encoder = OpenAIEncoder(model_name, **kwargs)
+        except OpenAIError as e:
+            raise RuntimeError(
```

Comment:
Probably for the future, but I guess it's reasonable to have our own auth and rate limit errors. For many services we are going to use this issue might occur, even pinecone by itself

Reply:
Yeah, that's definitely the plan!